sdk/core/core-rest-pipeline/src/policies/proxyPolicy.ts

```diff
@@ -38,6 +32,15 @@ const globalBypassedMap: Map<string, boolean> = new Map();
 let httpsProxyAgent: https.Agent | undefined;
 let httpProxyAgent: http.Agent | undefined;
 
+/**
+ * A function to reset the cached agents.
+ * @internal
+ */
+export function resetCachedProxyAgents(): void {
```

Comment:
I'm wondering if we should instead cache the agents at the policy instance level instead of in the module - it seems a little odd that I can create a fresh proxyPolicy instance but get the old cache. Then we wouldn't have to worry about clearing this for tests.

Reply:
Agree that caching proxy agent at the policy is more correct.